src/openms/include/OpenMS/FORMAT/HANDLERS/MzMLHandler.h

```diff
@@ -1492,6 +1533,7 @@ namespace OpenMS
         if (spectrum_data_.size() >= options_.getMaxDataPoolSize())
         {
           populateSpectraWithData();
+          spectrum_data_.reserve(options_.getMaxDataPoolSize());
```

Comment:
why is this done?
Line 1533 already establishes that the capacity is large enough.
populateSpectraWithData(); calls clear(), but that will not shrink the vector.